tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+  provides:
+    - C_VIO:
+    - A_IO9_I:
+    - A_IO10_O:
+    - C_VIO:
+    - STDOUT:
+    - STDIN:
+    - STDERR:
+    - Heap: 65536
+
+target-types:
+  - type: NUCLEO-G474RE
+    board:NUCLEO-G474RE
+
+components:
+  component: CMSIS:CORE
+  component: CMSIS Driver:USART:Custom
+  component: CMSIS Driver:VIO:Board&NUCLEO-G474RE
+  component: Compiler&ARM Compiler:Event Recorder&DAP
+    # Config/EventRecorderConf.h version="1.1.0"
+  component: Compiler&ARM Compiler:I/O:STDERR&User
+  component: Compiler&ARM Compiler:I/O:STDIN&User
+  component: Compiler&ARM Compiler:I/O:STDOUT&User
+  component: Board Support&NUCLEO-G474RE:Drivers:Basic I/O
+    # Drivers/Config/stm32g4xx_nucleo_conf.h version="1.0.0"
+  component: Device&STM32CubeMX:STM32Cube Framework:STM32CubeMX
+  component: Device&STM32CubeMX:STM32Cube HAL
+  component: Device&STM32CubeMX:Startup
+
+groups:
+  - group: Board IO
+    files:
+      - file: ./Board_IO/arduino.c
+      - file: ./Board_IO/arduino.h
+      - file: ./Board_IO/retarget_stdio.c
+  - group: STM32CubeMX
+    files:
+      - file: ./RTE/Device/STM32G474RETx/STCubeGenerated/STCubeGenerated.ioc
+```
+
+
+
+Todo: work on this
+
+The YML structure of the section `layer:` is:
+```yml
+layer:
+  description:
+  ....
+```  
+
+
+# Proposals
+
+## PLM of configuration file
+
+Configuration files require Project Lifetime Management (PLM).  This is a proposal how to manage different versions:
+
+Let’s assume a configuration file named “MyConfigFile.h” with version 1.2.0 which is part of component “MyComponent” and the destination directory is RTE/MyComponentClass/
+
+1. “MyComponent” is selected by the project/layer
+if `RTE/MyComponentClass/MyConfigFile.h` does not exist `MyConfigFile.h` will be copied from the pack where the component version of MyComponent is present `RTE/MyComponentClass/MyConfigFile.h` in addition a copy of this file will be created `RTE/MyComponentClass/.MyConfigFile.h-1.2.0`. The user will only see and edit the `MyConfigFile.h`.
+
+2. If a new version of the component with a new version `1.3.0` of the config file becomes available the tools are able to:
+    - read from the filename `.MyConfigFile.h-1.2.0` that the current file is based on version 1.2.0 
+    - that the new configfile `1.3.0` is greater in the minimum version only -> new feature
+    - if the tools run a simple diff on `MyConfigFile.h` and `.MyConfigFile.h-1.2.0` “local” modifications can be identified with
+         - (a) print a user notification about the config file not being up-to-date
+         - (b) run a 3-way merge utility attempting to merge the files if it fails revert back.
+
+
+## Output versions to *.cprj
+
+The ProjMgr should always generate *.cprj files that contain version information about the used software packs.
+
+## CMSIS-Zone Integration
+
+Add to the project the possiblity to specify.  The issue might be that the project files become overwelming, alternative is to keep partitioning in separate files.
```

Comment:
could be in separate file, but we should find a way to link this external file with he solution's projects it applies to